### DISCUSSION TITLE
How to adjust the size of pyplot on-the-fly

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#1731] [Q&A] [answered]
### Question

In this example:
```python
import numpy as np
from matplotlib import pyplot as plt
from nicegui import ui

with ui.pyplot(figsize=(3, 2)):
    x = np.linspace(0.0, 5.0)
    y = np.cos(2 * np.pi * x) * np.exp(-x)
    plt.plot(x, y, '-')

ui.run()
```
I would like the figsize of the plot to be adjusted on the fly, depending on the size of a browser, e.g. users may adjust the browser window. How to achieve this?

**Comment by falkoschindler** [comment, marked as answer]
This is a bit tricky, but here is a solution:

```py
import numpy as np
from matplotlib import pyplot as plt
from nicegui import ui

@ui.refreshable
def plot(page_width: int = 0) -> None:
    if page_width == 0:
        return
    with ui.pyplot(figsize=(page_width / 100, page_width / 150)):
        x = np.linspace(0.0, 5.0)
        y = np.cos(2 * np.pi * x) * np.exp(-x)
        plt.plot(x, y, '-')

plot()

resize = ui.button().on('resize', lambda e: plot.refresh(e.args), throttle=0.05).classes('hidden')
ui.add_head_html(f'''
<script>
function emitResizeEvent() {{
    content = document.querySelector('.nicegui-content');
    getElement({resize.id}).$emit('resize', content.offsetWidth);
}}
window.onload = emitResizeEvent;
window.onresize = emitResizeEvent;
</script>
''')
```

- The plot is wrapped in a `ui.refreshable` for easy re-rendering. Note that this will completely remove the plot and create a new one whenever the width changes.
- In order to register a custom event, we need a Vue element. Here I chose a `ui.button`, but it could be another one as well. It is hidden anyway. (Somehow `ui.element` doesn't work... We should think about a better way for registering such "global" events. But for now it's working.)
- The custom JavaScript code registers the `onload` and `onresize` events and emits the current width of the `.nicegui-content` div. You might want to chose some other metric here.
- The `plot()` function consumes the `page_width` and computes the `figsize` accordingly.
- The initial `plot()` call is needed to create the refreshable container, but exits early because the page width is still unknown.
- In order to same some computation and bandwidth, the event is throttled with 0.05s.

**Reply by nghia-vo** [reply]
Thanks very much. I'll incorporate these to my app and give feedbacks.
